_problems/fee-bumping.md

```diff
@@ -0,0 +1,96 @@
+
+It is much more desirable to allow the party to spontaneously increase the fee.
+This can be done with [CPFP].
+For example, in the [Lightning] protocol commitment transactions are designed with [anchor outputs] so they can be easily fee bumped through CPFP.
+
+### Resource Usage of Replacement
+
+Allowing a mempool transaction to be replaced by higher fee paying transactions is that this may open up nodes to *denial of service* (DoS) attacks.
+Without replacement a malicious node's ability to get your node to process a transaction is limited by the number of UTXOs they own.
+With replacement an attacker can repeatedly spam your node with transactions that replace previous ones even if they only own a single UTXO.
+
+To prevent effective denial of service attacks, replacement transactions have to follow some rather strict rules as defined in [BIP125].
+In particular a replacement transaction must pay a higher absolute fee and a higher feerate than the transactions they are evicting and it must not evict more than 100 transactions.
+
+For layer-2 protocols this rules had the unintended consequence of introducing  "pinning" attacks (see [Related Research](#related-research)).
+This is usually done by adding a lot of descendants to a transaction that conflicts with the one the honest party needs confirm.
+As we have said an honest party needs to confirm a transactions in a certain time -- if it can't evict a maliciously crafted transaction then they are in a helpless position.
+
+An instinctive exception to [BIP125]'s rules called the [CPFP carve out] was introduced to allow lightning designers to avoid pinning attacks with some extra design effort.
+
+## Impact
+
+The principles layer-2 protocol designers should use to enable transactions to confirm on time are an esoteric domain understood by only a handful of people.
+Even those practiced in the occult [have made mistakes](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-September/002796.html) trying to design around them.
+Having a simple set of principles would increase the confidence in the resulting protocols and allow a wider group to understand and build layer-2 systems.
+
+## Potential Directions
+
+1. **Reconsidering mempool and relay rules**: It is possible that mempool rules could be made less problematic for layer-2 protocol designers without harming the Bitcoin system.
+   The main concern with this approach is that the rules that are optimal for layer-2 protocols may not be the rules that are optimal for miner profit or preventing DoS attacks.
+   However, if tweaked relay rules were to allow layer-2 Bitcoin protocols to thrive then the increased value of the Bitcoin network could more than compensate miners for using a slightly suboptimal algorithm (as long as they don't introduce DoS attacks).
```

Comment:
I believe that a mempool acceptance rules revamp can still combine security mechanism for L2s and miner incentives. In anyway, fee-bumping is good for miners as you're making the blockspace demand more competitive.

Reply:
I agree and I think this concern can be dismissed in a different way: per-utxo transactions fee optimization is not important for miners as it will create negligible revenue. It is about making sure they have the highest fee paying transaction set from different utxos (i.e. different users are competing for blockspace). There is no need to necessarily resolve two transactions that share the same input strictly in favor of the one that pays the highest fee in absolute terms of feerate. Or at least DoS is the only concern here not revenue.

Reply:
I'm curious about more refined thoughts on this issue as I think we might have to switch from replace-by-fee to replace-by-feerate to solve some pinnings.

Further, having a consistent miner model w.r.t to conflicting transactions is an interesting research, likely deserves its own page too.

Reply:
> I'm curious about more refined thoughts on this issue as I think we might have to switch from replace-by-fee to replace-by-feerate to solve some pinnings.

Actually this was going to be the gist of my commentary! Along with how to better deal with the mass descendants eviction problem than BIP125.

> Further, having a consistent miner model w.r.t to conflicting transactions is an interesting research, likely deserves its own page too.

Is this actually that deep of a topic? I'd say unless it turns out that there is a lot more than meets the eye then it's part of this.